tasks/setup_debian.yml

```diff
@@ -10,28 +10,62 @@
         state: present
         update_cache: true
 
-    - name: Install mariadb repository key
+    - name: Ensure keyrings dir exists
+      ansible.builtin.file:
+        path: /etc/apt/keyrings
+        state: directory
+        owner: root
+        group: root
+        mode: "0755"
+
+    - name: Fetch mariadb repository key
+      # change is always triggered in check mode
+      when: not ansible_check_mode
       ansible.builtin.get_url:
-        url: "https://mariadb.org/mariadb_release_signing_key.asc"
-        dest: "/etc/apt/trusted.gpg.d/mariadb.asc"
+        url: https://mariadb.org/mariadb_release_signing_key.asc
+        dest: /etc/apt/keyrings/mariadb-archive-keyring.asc
         # //TEMP todo
         # checksum: sha256:http://example.com/path/sha256sum.txt
         owner: root
         group: root
         mode: "0644"
-      # change is always triggered in check mode
-      when: not ansible_check_mode
+      register: fetch_repository_key
+
+    - name: Dearmor mariadb repository key
+      when: fetch_repository_key is changed
+      ansible.builtin.command: >
+        gpg --dearmor
+        -o /etc/apt/keyrings/mariadb-archive-keyring.gpg
+        /etc/apt/keyrings/mariadb-archive-keyring.asc
 
-    - name: Determine distribution name (Debian sid pb)
+    - name: Determine repo distribution release
       ansible.builtin.set_fact:
-        distribution: "{% if 'n/a' in ansible_distribution_release %}sid{% else %}{{ ansible_distribution_release }}{% endif %}"
+        repo_distribution_release: >-
+          {{ 'sid' if 'n/a' in ansible_distribution_version else ansible_distribution_release }}
 
     - name: Setup mariadb repository sourcelist entry
+      when: not mariadb_use_official_repo_deb822
       ansible.builtin.copy:
         dest: /etc/apt/sources.list.d/mariadb.list
         content: |
           # Ansible managed
-          deb {{ mariadb_use_official_repo_url }}/{{ mariadb_use_official_repo_version }}/{{ ansible_distribution | lower() }} {{ distribution }} main
+          deb [signed-by=/etc/apt/keyrings/mariadb-archive-keyring.gpg] {{ mariadb_use_official_repo_url }}/{{ mariadb_use_official_repo_version }}/{{ ansible_distribution | lower() }} {{ repo_distribution_release }} main
+        owner: root
+        group: root
+        mode: "0644"
+
+    - name: Setup mariadb repository sources entry
+      when: mariadb_use_official_repo_deb822
+      ansible.builtin.copy:
+        dest: /etc/apt/sources.list.d/mariadb.sources
+        content: |
+          # Ansible managed
+          X-Repolib-Name: MariaDB
+          Types: deb
+          URIs: {{ mariadb_use_official_repo_url }}/{{ mariadb_use_official_repo_version }}/{{ ansible_distribution | lower() }}
+          Suites: {{ repo_distribution_release }}
+          Components: main
+          Signed-By: /etc/apt/keyrings/mariadb-archive-keyring.gpg
```

Comment:
Can you change for `mariadb-keyring.gpg`, small is beautiful :)

Reply:
I am following the debian keyrings names, but sure.

![image](https://user-images.githubusercontent.com/19195485/208127902-12f6d9fa-3325-4655-966f-0e012106339d.png)

Reply:
According to https://wiki.debian.org/DebianRepository/UseThirdParty#OpenPGP_certificate_distribution we diverged a bit with regards to the naming, unless we want to be maniac about this, I would say we are fine with the shorter name.